lib/firespring_dev_commands/aws/route53.rb

```diff
@@ -0,0 +1,72 @@
+require 'aws-sdk-route53'
+
+module Dev
+  class Aws
+    # Class for performing Route53 functions
+    class Route53
+      attr_reader :client, :zones
+
+      def initialize
+        @client = ::Aws::Route53::Client.new
+        @zones = []
+      end
+
+      def get_hosted_zones(domains)
+        if domains.empty?
+          response = client.list_hosted_zones
+          response.hosted_zones.each do |hosted_zone|
+            @zones << hosted_zone.id
+          end
+        else
+          domains.each do |domain_name|
+            zone = client.list_hosted_zones_by_name({dns_name: domain_name, max_items: 1})
+            target_name = zone.hosted_zones[0].name.chomp!('.') if zone.hosted_zones[0].name.end_with?('.')
+            @zones << zone.hosted_zones[0].id unless target_name != domain_name
+          end
+        end
+        raise 'Hosted zone(s) not found.' if @zones.empty?
+      end
+
+      def get_target_config_id(zone_id)
+        config = client.list_query_logging_configs(
+          hosted_zone_id: zone_id,
+          max_results: '1'
+        )
```

Comment:
I'd recommend re-writing this like

```
 def get_target_config_id(zone_id)
        client.list_query_logging_configs(
          hosted_zone_id: zone_id,
          max_results: '1'
        ).query_logging_configs&.first&.id
      end
```

I don't think verifying the hosted zone id is necessary since we included it in the request?